addOns/network/src/main/java/org/zaproxy/addon/network/internal/client/apachev5/HttpSenderApache.java

```diff
@@ -415,7 +419,34 @@ public InputStream getResponseBodyAsStream() {
                     };
             method.setUpgradedSocket(socket);
             method.setUpgradedInputStream(socket.getInputStream());
+            Object userObject = message.getUserObject();
             message.setUserObject(method);
+
+            if (isPersistentManualConnection(userObject)
+                    && !legacyProxyListenerHandler
+                            .get()
+                            .notifyPersistentConnectionListener(message, null, method)) {
```

Comment:
Worth adding a unit test?

Reply:
Always but I don't want to right now.

Reply:
Tests added.

Reply:
I was totally ready to let it go.

Thanks for tackling that!